app/src/main/kotlin/com/hedvig/app/feature/marketing/data/UpdateApplicationLanguageUseCase.kt

```diff
@@ -1,19 +1,16 @@
 package com.hedvig.app.feature.marketing.data
 
-import android.content.Context
+import androidx.appcompat.app.AppCompatDelegate
+import androidx.core.os.LocaleListCompat
 import com.hedvig.android.market.Language
 import com.hedvig.android.market.Market
 import com.hedvig.android.market.MarketManager
-import com.hedvig.app.feature.marketpicker.LocaleBroadcastManager
 
 class UpdateApplicationLanguageUseCase(
   private val marketManager: MarketManager,
-  private val localeBroadcastManager: LocaleBroadcastManager,
-  private val context: Context,
 ) {
   operator fun invoke(market: Market, language: Language) {
     marketManager.market = market
-    Language.persist(context, language)
-    localeBroadcastManager.sendBroadcast()
+    AppCompatDelegate.setApplicationLocales(LocaleListCompat.forLanguageTags(language.toString()))
```

Comment:
This happens both here and inside the LanguageService. Maybe make this use-case have a dep on it and call that instead?